service/controller/resource/monitoring/prometheus/node_count.go

```diff
@@ -0,0 +1,86 @@
+package prometheus
+
+import (
+	"github.com/giantswarm/k8sclient/v7/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	expv1beta1 "sigs.k8s.io/cluster-api/exp/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/prometheus-meta-operator/v2/service/key"
+)
+
+func getClusterNodeCount(ctx context.Context, k8sClient k8sclient.Interface, cluster metav1.Object) (int, error) {
+	mdCount, err := getClusterNodeCountFromMachineDeployment(ctx, k8sClient, cluster)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	mpCount, err := getClusterNodeCountFromMachinePool(ctx, k8sClient, cluster)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	var nodeCount int32 = 1
+	if mdCount > nodeCount {
+		nodeCount = mdCount
+	}
+	if mpCount > nodeCount {
+		nodeCount = mpCount
+	}
+
+	// TODO: Is this ok ? Are we loosing precision here going from int32 to int ?
+	return int(nodeCount), nil
+}
+
+func getClusterNodeCountFromMachineDeployment(ctx context.Context, k8sClient k8sclient.Interface, cluster metav1.Object) (int32, error) {
+	var mds apiv1beta1.MachineDeploymentList
+
+	var opts []client.ListOption
+	opts = append(opts, client.MatchingLabels{
+		key.ClusterLabel: key.ClusterID(cluster),
+	})
+	opts = append(opts, client.InNamespace(key.OrganizationNamespace(cluster)))
+
+	err := k8sClient.CtrlClient().List(ctx, &mds, opts...)
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	var highestNodeCount int32 = 0
```

Comment:
Same I think we do not want to know the highest node count but sum of all the nodes

Reply:
This is a best effort at trying to find the node count, since we do not really know where the information is I am trying to find the highest value. I think some provider might have both MachineDeployment and MachinePool, I am not sure what adding those values would mean. Needs to be tested.

Reply:
The thing here is that I do not understand what you want to achieve tbh. If you want to count all the nodes, you have to sum the number of nodes for all machine pools of a cluster (because customers can have multiple node pools for different reasons) + the number of nodes for all the machine deployments so we can correlate the memory per node.

What you are getting right now is the highest value of the bigger machine pool or machine deployment but if customers have 5 machine pools of 20 nodes each, using 20 as a base value feels wrong.

For CAPI, you will have to ensure you do not fetch the control plane node machine pool/deployment.